src/common/extensionMessaging.ts

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+
+import * as Q from "q";
+import * as http from "http";
+
+/**
+ * Message server parameters.
+ */
+export let ServerParams = {
+    PORT: 8099,
+    HOST: "127.0.0.1"
+};
+
+/**
+ * Defines the messages sent to the extension.
+ * Add new messages to this enum.
+ */
+export enum ExtensionIncomingMessage {
+    START_PACKAGER,
+    STOP_PACKAGER
+}
+
+/**
+ * Generic interface for messages with arguments.
+ */
+export interface MessageWithArgs<T> {
+    message: T;
+    args?: any[];
+}
+
+/**
+ * Sends messages to the extension.
+ */
+export class ExtensionMessageSender {
+
+    public sendMessage(message: MessageWithArgs<ExtensionIncomingMessage>): Q.Promise<any> {
+        let deferred = Q.defer<any>();
+
+        let options = {
+            host: ServerParams.HOST,
+            port: ServerParams.PORT,
+            path: "/",
```

Comment:
My first instinct is that we should pick an approach and go with it: Either a socket protocol of sorts with particular formatted messages on the wire, or a HTTP protocol where we use URLs and the body of the request is minimal.